src/main/java/com/spotify/confidence/FlagResolverImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.spotify.confidence;
+
+import com.google.common.base.Strings;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.protobuf.Struct;
+import com.spotify.confidence.shaded.flags.resolver.v1.*;
+import io.grpc.ManagedChannel;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+class FlagResolverImpl implements FlagResolver, Closeable {
+
+  // Deadline in seconds
+  public static final int DEADLINE_AFTER_SECONDS = 10;
+  private final ManagedChannel managedChannel;
+
+  private final String clientSecret;
+
+  private final String SDK_VERSION;
+  private static final SdkId SDK_ID = SdkId.SDK_ID_JAVA_PROVIDER;
+
+  private final FlagResolverServiceGrpc.FlagResolverServiceFutureStub stub;
+
+  public FlagResolverImpl(String clientSecret, ManagedChannel managedChannel) {
+    this.clientSecret = clientSecret;
+    this.managedChannel = managedChannel;
+    this.stub = FlagResolverServiceGrpc.newFutureStub(managedChannel);
+
+    if (Strings.isNullOrEmpty(clientSecret)) {
+      throw new IllegalArgumentException("clientSecret must be a non-empty string.");
+    }
+
+    try {
+      final Properties prop = new Properties();
+      prop.load(this.getClass().getResourceAsStream("/version.properties"));
+      this.SDK_VERSION = prop.getProperty("version");
+    } catch (IOException e) {
+      throw new RuntimeException("Can't determine version of the SDK", e);
+    }
+  }
+
+  public ListenableFuture<ResolveFlagsResponse> resolveFlags(
```

Comment:
I think this should rather return a CompletionStage? ListenableFuture is a Google type right?

Reply:
yeah, I've had trouble converting them in a nice way. But as long as this is internal I don't really care.

Reply:
I'm picking this one now.